controllers/postgres_controller.go

```diff
@@ -475,6 +469,51 @@ func (r *PostgresReconciler) updatePodEnvironmentConfigMap(ctx context.Context,
 	return nil
 }
 
+func (r *PostgresReconciler) updatePodEnvironmentSecret(ctx context.Context, p *pg.Postgres) error {
+	log := r.Log.WithValues("postgres", p.Name)
+	if p.Spec.BackupSecretRef == "" {
+		log.Info("No configured backupSecretRef found, skipping configuration of postgres backup")
+		return nil
+	}
+
+	backupConfig, err := r.getBackupConfig(ctx, p.Namespace, p.Spec.BackupSecretRef)
+	if err != nil {
+		return err
+	}
+
+	awsAccessKeyID := backupConfig.S3AccessKey
+	awsSecretAccessKey := backupConfig.S3SecretKey
+
+	// create updated content for pod environment configmap
+	data := map[string][]byte{
+		"AWS_ACCESS_KEY_ID":     []byte(awsAccessKeyID),
+		"AWS_SECRET_ACCESS_KEY": []byte(awsSecretAccessKey),
+	}
+
+	// libsodium client side encryption key // TODO validate in server-side that it is 32 bytes long, see https://github.com/wal-g/wal-g#encryption
+	if backupConfig.S3EncryptionKey != nil {
+		data["WALG_LIBSODIUM_KEY"] = []byte(*backupConfig.S3EncryptionKey)
+	}
+
+	s := &corev1.Secret{}
+	ns := types.NamespacedName{
+		Name:      operatormanager.PodEnvCMName,
+		Namespace: p.ToPeripheralResourceNamespace(),
+	}
+	if err := r.SvcClient.Get(ctx, ns, s); err != nil {
```

Comment:
Original error is swallowed, wouldn't it be cleaner to check for `apierrors.NotFound`? Also `CreatePodEnvironmentSecret` already checks if it already exists and returns the secret, so no need to it twice.